themes/default/content/blog/mlops-python-iac-deploy-a-chatbot-in-30-minutes/index.md

```diff
@@ -0,0 +1,197 @@
+
+Let's introduce some key concepts and technologies to start off.
+
+- Infrastructure as Code (IaC) is a transformative approach to platform engineering and orchestration. By applying the rigor and precision of software development practices to cloud operations, IaC offers AI/ML professionals an efficient, reliable, and predictable way to develop at the highest velocity.
+
+- Katwalk Server is a demo AI application written to host your choice of LLM either locally or in the cloud, and serve it as an OpenAI API compatible service. As a practical, hands-on introduction to IaC for the MLOps space, this project aims to demystify the concepts and showcase the benefits Pulumi can bring to your AI/ML projects.
+
+- Large Language Model or "LLM"s like ChatGPT are machine learning models trained to generate human-like conversational text which have already become essential tools for businesses and individuals, transforming every day tasks and how we interact with technology.
+
+> figure 2. Or just let ChatGPT describe itself:
+![Alt text](image-1.png)
+
+If you are ready to try the *#MLOpsChallenge* along with us, then let's double check a few requirements before starting the clock on our 30 minute chatbot challenge.
+
+## Requirements:
+
+* [Pulumi](https://www.pulumi.com/docs/install/)
+* [Python3](https://www.python.org/downloads/)
+* [Git CLI](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [Huggingface Token](https://huggingface.co/docs/transformers.js/guides/private)
+* [Huggingface access to LLaMa2](https://huggingface.co/meta-llama)
+  * [Meta LLaMa2 Access](https://ai.meta.com/resources/models-and-libraries/llama-downloads/)
+
+Optional Requirements (must choose one)
+
+* [Docker](https://docs.docker.com/engine/install/)
+  * [Nvidia CUDA Enabled GPU](https://developer.nvidia.com/cuda-gpus)
+  * [Nvidia Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)
+* [Azure](https://azure.microsoft.com/en-us)
+  * [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+* [Runpod.io](https://runpod.io)
+  * [Runpod api key](https://docs.runpod.io/docs/graphql-api)
+
+--------------------------------------------------------------------------
+
+Now, for the real fun, let's start the timer and deploy our own chatbot!
+
+### 1. Prepare your Pulumi IaC Directory:
+
+First we need to clone our IaC repository from Github and prepare our python environment.
+
+```bash
+# Clone the repository and cd to the pulumi iac directory
+gh repo clone usrbinkat/katwalk && cd katwalk/pulumi
+
+# Create and initialize the python virtual env
+python3 -m venv venv && source venv/bin/activate
+
+# Install python dependencies
+python -m pip install -r requirements.txt
+```
+
+### 2. Login & Initialize your pulumi state file and stack
+
+Next let's setup our secure configuration and secret storage in a local file for now.
```

Comment:
Is it worth including links to the Pulumi Secrets and State documentation? If not in this section then maybe towards the end as part of the call to action section.

Reply:
it is